Maddori.Apple/Maddori.Apple/Global/UIComponent/View/AlertViewController.swift

```diff
@@ -0,0 +1,223 @@
+        case .join:
+            return "팀에 합류하시겠어요?"
+        }
+    }
+    
+    var actionTitle: String {
+        switch self {
+        case .delete:
+            return "삭제"
+        case .join:
+            return "참여"
+        }
+    }
+    
+    var actionTitleColor: UIColor {
+        switch self {
+        case .delete:
+            return .red100
+        case .join:
+            return .gray500
+        }
+    }
+}
+
+final class AlertViewController: BaseViewController {
+    
+    var type: AlertType
+    var teamName: String? = nil
+    
+    init(type: AlertType) {
+        self.type = type
+        super.init()
+    }
+    
+    init(type: AlertType, teamName: String?) {
+        self.type = type
+        self.teamName = teamName
+        super.init()
+    }
+    
+    required init?(coder: NSCoder) { nil }
+    
+    private enum Size {
+        static let alertViewWidth: CGFloat = 265
+        static let alertViewHeight: CGFloat = 163
+        static let dividerWeight: CGFloat = 0.5
+        static let buttonMinimumHeight: CGFloat = 44
+    }
+    
+    // MARK: - property
+    
+    private let alertView: UIView = {
+        let view = UIView()
+        view.backgroundColor = .white100
+        view.layer.cornerRadius = 10
+        return view
+    }()
+    private lazy var titleLabel: UILabel = {
+        let label = UILabel()
+        label.text = type.title
+        label.textColor = .blue200
+        label.font = .main
+        return label
+    }()
+    private lazy var subTitleLabel: UILabel = {
+        let label = UILabel()
+        label.text = type.subTitle
+        label.textColor = .gray400
+        label.font = .caption1
+        return label
+    }()
+    private let labelButtonDivider: UIView = {
+        let view = UIView()
+        view.backgroundColor = .gray300
+        return view
+    }()
+    private let buttonIntervalDivider: UIView = {
+        let view = UIView()
+        view.backgroundColor = .gray300
+        return view
+    }()
+    private lazy var cancelButton: UIButton = {
+        let button = UIButton()
+        button.setTitle(TextLiteral.alertViewCancelButtonText, for: .normal)
+        button.setTitleColor(.gray500, for: .normal)
+        button.titleLabel?.font = .caption2
+        button.titleLabel?.textAlignment = .center
+        let action = UIAction { [weak self] _ in
+            self?.dismiss(animated: true)
+        }
+        button.addAction(action, for: .touchUpInside)
+        return button
+    }()
+    private lazy var actionButton: UIButton = {
+        let button = UIButton()
+        button.setTitle(type.actionTitle, for: .normal)
+        button.setTitleColor(type.actionTitleColor, for: .normal)
+        button.titleLabel?.font = .caption2
+        button.titleLabel?.textAlignment = .center
+        let action = UIAction { [weak self] _ in
+            self?.didTappedActionButton(self!.type)
```

Comment:
p3;

self 를 일단은 강제로 풀긴했는데 풀어서 넣어주는게 좋아보여요! 

방법은 뭐가 있을까용... 으므으음

Reply:
```suggestion
        let action = UIAction { [weak self] _ in
            if let type = self?.type {
                self?.didTappedActionButton(type)
            }
        }
```
`if let`으로 풀 수 있겠군요 !

Reply:
[61ecb1a](https://github.com/DeveloperAcademy-POSTECH/MacC-Team-Maddori.Apple/pull/78/commits/61ecb1a66f226248d4a7d20d9f2bd08daacd30fa)
감사합니당! 체크리스트 업데이트 ,,,, ⭐️
[✔️] ! 사용하기보다는 if let 이용해서 ! 풀어주기 > 값이 없으면 빌드 에러가 난다!!!!!